Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc

```diff
@@ -214,6 +224,186 @@ bool HGCalParametersFromDD::build(const DDCompactView* cpv,
   return ok;
 }
 
+bool HGCalParametersFromDD::build(const cms::DDCompactView* cpv,
+                                  HGCalParameters& php,
+                                  const std::string& name,
+                                  const std::string& namew,
+                                  const std::string& namec,
+                                  const std::string& namet,
+                                  const std::string& name2) {
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "HGCalParametersFromDD (DD4Hep)::build called with "
+                                << "names " << name << ":" << namew << ":" << namec << ":" << namet;
+#endif
+  cms::DDFilteredView fv(cpv->detector(), cpv->detector()->worldVolume());
+  cms::DDVectorsMap vmap = cpv->detector()->vectors();
+  std::vector<std::string> tempS;
+  std::vector<double> tempD;
+  tempS = fv.get<std::vector<std::string> >(name.c_str(), "GeometryMode");
+  std::string sv = (!tempS.empty()) ? tempS[0] : "HGCalGeometryMode::Hexagon8Full";
+  std::string attribute = "Volume";
+  cms::DDSpecParRefs refs;
+  const cms::DDSpecParRegistry& mypar = cpv->specpars();
+  mypar.filter(refs, attribute, name);
+  fv.mergedSpecifics(refs);
+  bool ok = fv.firstChild();
+  HGCalGeometryMode::WaferMode mode(HGCalGeometryMode::Polyhedra);
+
+  if (ok) {
+    php.mode_ = getGeometryMode(sv);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "GeometryMode " << php.mode_ << ":" << HGCalGeometryMode::Hexagon << ":"
+                                  << HGCalGeometryMode::HexagonFull << ":"
+                                  << ":" << HGCalGeometryMode::Hexagon8 << ":" << HGCalGeometryMode::Hexagon8Full << ":"
+                                  << ":" << HGCalGeometryMode::Trapezoid;
+#endif
+    php.levelZSide_ = 3;        // Default level for ZSide
+    php.detectorType_ = 0;      // These two parameters are
+    php.firstMixedLayer_ = -1;  // defined for post TDR geometry
+    HGCalGeomParameters* geom = new HGCalGeomParameters();
```

Comment:
this looks like a memory leak - better use a `unique_ptr`

Reply:
Done

Reply:
@bsunanda - it looks like it's not updated.